implementation/src/main/java/io/smallrye/config/EnvConfigSource.java

```diff
@@ -110,6 +121,38 @@ private static String replaceNonAlphanumericByUnderscores(String name) {
         return sb.toString();
     }
 
+    private static String envToProperty(final String name) {
+        int length = name.length();
```

Comment:
I am a bit confused why it is so complicated with quotesOpen logic. What kind of scenario do you try to accomplish?

Reply:
We have cases where quotes are used in property names, so something like `my."property".something` the equivalent for the env variable is `MY__PROPERTY__SOMETHING`.